ground/src/main/java/com/google/android/ground/persistence/remote/firebase/FirebaseFirestoreProvider.kt

```diff
@@ -28,11 +28,13 @@ import timber.log.Timber
 class FirebaseFirestoreProvider @Inject constructor(settings: FirebaseFirestoreSettings) :
   AsyncSingletonProvider<FirebaseFirestore>({
     FirebaseFirestore.getInstance().also {
-      try {
-        it.firestoreSettings = settings
-      } catch (e: IllegalStateException) {
-        // Logging added for #2377.
-        Timber.e(e, "Singleton provider likely initialized multiple times")
+      if (it.firestoreSettings != settings) {
```

Comment:
Instead of nesting, can we return early?

Reply:
Done